[BUILD] Build break in OLTP_FILE tests

Using CMake -DBUILD_TESTING=ON -DWITH_OTLP_FILE=ON

The build fails with:

```
[ 83%] Building CXX object exporters/otlp/CMakeFiles/otlp_file_client_test.dir/test/otlp_file_client_test.cc.o
/.../opentelemetry-cpp/exporters/otlp/test/otlp_file_client_test.cc:23:10: fatal error: nlohmann/json.hpp: No such file or directory
   23 | #include "nlohmann/json.hpp"
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [exporters/otlp/CMakeFiles/otlp_file_client_test.dir/build.make:76: exporters/otlp/CMakeFiles/otlp_file_client_test.dir/test/otlp_file_client_test.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6643: exporters/otlp/CMakeFiles/otlp_file_client_test.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Note that nlohmann_json is not installed under /usr/include, but in a different location, so it is not part of the default include path.

--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -489,8 +489,11 @@ if(BUILD_TESTING)
   if(WITH_OTLP_FILE)
     add_executable(otlp_file_client_test test/otlp_file_client_test.cc)
     target_link_libraries(
-      otlp_file_client_test ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
-      ${GMOCK_LIB} opentelemetry_exporter_otlp_file
+      otlp_file_client_test
+      ${GTEST_BOTH_LIBRARIES}
+      ${CMAKE_THREAD_LIBS_INIT}
+      ${GMOCK_LIB}
+      opentelemetry_exporter_otlp_file
       opentelemetry_otlp_recordable
       nlohmann_json::nlohmann_json)
     gtest_add_tests(

--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -491,7 +491,8 @@ if(BUILD_TESTING)
     target_link_libraries(
       otlp_file_client_test ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
       ${GMOCK_LIB} opentelemetry_exporter_otlp_file
-      opentelemetry_otlp_recordable)
+      opentelemetry_otlp_recordable
+      nlohmann_json::nlohmann_json)
     gtest_add_tests(
       TARGET otlp_file_client_test
       TEST_PREFIX exporter.otlp.

